src/common/io.cc

```diff
@@ -369,7 +369,7 @@ int64_t read_vector(int fd, int64_t *type, std::vector<uint8_t> &buffer) {
   if (closed) {
     goto disconnected;
   }
-  if (length > buffer.size()) {
+  if (static_cast<size_t>(length) > buffer.size()) {
     buffer.resize(length);
```

Comment:
This would "work" but it's very confusing and wouldn't make sense if length is negative. There should at least be a cast here. But see my comment above.